tests/staticfiles_tests/test_checks.py

```diff
@@ -132,3 +132,65 @@ def test_nonexistent_directories(self):
             # Nonexistent directories are skipped.
             finder = get_finder("django.contrib.staticfiles.finders.FileSystemFinder")
             self.assertEqual(list(finder.list(None)), [])
+
+
+class StorageCheckTests(SimpleTestCase):
+    def test_no_errors_without_setting(self):
+        errors = check_storages(None)
+        self.assertEqual(errors, [])
+
+    @override_settings(
+        STORAGES={
+            "default": {
+                "BACKEND": "django.core.files.storage.FileSystemStorage",
+            },
+            "staticfiles": {
+                "BACKEND": "django.contrib.staticfiles.storage.StaticFilesStorage",
+            },
+        }
+    )
+    def test_no_errors_staticfiles_and_default(self):
+        errors = check_storages(None)
+        self.assertEqual(errors, [])
+
+    @override_settings(
+        STORAGES={
+            "staticfiles": {
+                "BACKEND": "django.contrib.staticfiles.storage.StaticFilesStorage",
+            },
+        }
+    )
+    def test_no_errors_staticfiles_only(self):
+        # Despite the STORAGES setting only containing "staticfiles" in the above
+        # override_settings call, we need to remove it again with this .pop() call.
+        # This is because UserSettingsHolder has some backwards compatible code in
+        # __setattr__ to keep the "default" storage, which will be removed in 5.1.
+        # RemovedInDjango51Warning
```

Comment:
Here is the code that will be removed in 5.1:

https://github.com/django/django/blob/6b965c600054f970bdf94017ecf2e0e6e0a4326b/django/conf/__init__.py#L295-L318

When the code is removed, I think we won't need this `.pop()` anymore.

Reply:
Correct. I reduced the comment a bit and I referenced it from the following tests.